napari/utils/translations.py

```diff
@@ -381,13 +493,38 @@ def _np(self, msgctxt: str, msgid: str, msgid_plural: str, n: str) -> str:
             The plural string to translate.
         n : int
             The number for pluralization.
+        deferred : bool, optional
+            Define if the string translation should be deferred or executed
+            in place. Default is False.
+        kwargs : dict, optional
+            Any additional arguments to use when formating the string.
 
         Returns
         -------
-        str
-            The translated string.
+        TranslationString or str
+            The translation string which might be deferred or translated in
+            place.
         """
-        return self.npgettext(msgctxt, msgid, msgid_plural, n)
+        if deferred:
+            TranslationString(
+                domain=self._domain,
+                msgctxt=msgctxt,
+                msgid=msgid,
+                msgid_plural=msgid_plural,
+                n=n,
+                deferred=deferred,
+                **kwargs,
+            )
```

Comment:
No assignment to the variable. It looks like lack of test for this PR.

Reply:
Fixed and added tests!